### ISSUE TITLE
Provide UI to indicate available code actions (💡)

### ISSUE THREAD

**Issue by matklad** [mawww/kakoune#4387]
### Feature

:wave: I am matklad, on of the developers of rust-analyzer, a language server for Rust. 
I don't use kakoune myself, so sorry if I am missing something obvious.

One of the most important features of IDEs/LSP is the light-bulb/code action workflow. 
The idea here is that if IDE thinks there's a semantic action available *in the current* context, it shows an :bulb: icon.
When the user sees this icon, they can press a key to trigger one of the actions from the context menu. 
In a sense, this is an improvement over usual command pallet UX which adds context into the mix to filter the actions.
(see [this post](https://rust-analyzer.github.io/blog/2020/09/28/how-to-make-a-light-bulb.html) for even more context). 

I've noticed that at the moment kakoune doesn't actually display any indications that code actions are available. 
I feel pretty strongly that it should do that, as a lot of IDE features are hidden behind that UX :-)

Kakoune in default config:

![image](https://user-images.githubusercontent.com/1711539/137342972-7e51a213-ee01-4216-aac9-461ad532aea2.png)

VS Code

![ca](https://user-images.githubusercontent.com/1711539/137343859-9146b395-d37b-4b4d-a6f3-c64762bf3a08.gif)

Opening this against core editor repository, rather than kak-lsp, because architecturally :bulb: is usually an editor feature, much like completion widget.

**Comment by dpc** [comment]
> A user has asked to show all code actions for the buffer, not just at the cursor.

I'd also prefer that. Some very delicate and unobtrusive indication would be great to have.

>  cursor position is the input to the "get code actions" LSP requests.

One could just in the background request code actions in position of every word/token in the code, as a hack/workaround.

Alternatively, I'd guess that the door are open for `rust-analyzer` to champion an extension method that `kak-lsp` could be the first user of? :)

Anyway, for started being able to see the code action hint just like in VSCode would be great.

**Comment by matklad** [comment]
>I think Kakoune already has enough features to implement this.

One tiny push-back I have here is that putting the logic to do that into kak-lsp means that, if you want to display :bulb: in the editor, you have to use LSP. Ie, I can't use a non-LSP using thing to plug into :bulb: UI. So, it does seem to me that having something along `completers` but for :bulb: makes sense. But it definitelly makes sense to start doing it in the LSP, even if we'd love to have the UX in the editor proper at some point.

Side note: the terminology around :bulb: is all other the place -- IntelliJ calls them "intentions", VS Code -- "code actions". I personally think that the best term is the one from dart analyzer -- "assists".

**Comment by krobelus** [comment]
Yeah if multiple plugins actually want this, then an `assists` option akin to `completers` could work:
For example

```
set-option -add global assists option=my_assists
declare-option str-list my_assists
set-option buffer my_assists \
    'Add ";"' 'execute-keys -draft 123g<a-l>a<semicolon>' \
    'Reflow paragraph' 'execute-keys -draft <a-p>|fmt<ret>'
```

This would need three features:
1. Display the :bulb: whenever %opt{assists} changes and is non-empty
2. A command to execute a named assist - `run-assist 'Add ";"'`
3. A command to select an assist to run from %opt{assists} - here that's just `menu %opt{my_assists}`

Which plugins (including the ones that ship with Kakoune) would use that?

**Comment by matklad** [comment]
>Which plugins (including the ones that ship with Kakoune) would use that?

One possible choice is spell-checking. Rather than having to know and remember the `spell-replace` command, it can be an assist if the cursor is on a misspelled word
